### PR TITLE
Fix the PIR potential attacks

### PR DESCRIPTION
**Pull request by arvid220u**

### CODE REVIEW

File: daemon/transmitter/transmitter.cc [hunk 1]
```diff
@@ -24,8 +24,11 @@ auto generate_dummy_address(const db::Registration& reg) -> db::Address {
   // convert dummy_read_write_keys to a rust::Vec<uint8_t>
   auto read_key_vec_rust = string_to_rust_u8Vec(dummy_read_write_keys.first);
   auto write_key_vec_rust = string_to_rust_u8Vec(dummy_read_write_keys.second);
-
-  return (db::Address){-1, 0, read_key_vec_rust, write_key_vec_rust, 0};
+  // The dummy PIR index is designed to not coincide
+  // with any other index. Since it is very very
+  // large, it will result in an all zero PIR query.
```
Comment (arvid220u) [inline daemon/transmitter/transmitter.cc:29]: see my comment in asphr, i'm unsure why we are doing this

Comment (arvid220u) [reply daemon/transmitter/transmitter.cc:29]: i do like that it is a constant instead of magic number

**Review by arvid220u — COMMENTED** [review]
do we want to change the registration method also to not store the pir_secret_key and pir_galois_key?

or maybe let's just add a comment to say that these fields are now deprecated

otherwise lgtm

File: daemon/transmitter/transmitter.cc (hunk 1 above)
Comment (sualehasif) [reply daemon/transmitter/transmitter.cc:29]: added the comment there

**Pull request merged**